translator/src/transaction.rs

```diff
@@ -46,16 +46,26 @@ impl TelosEVMTransaction {
         let tx_raw = &mut raw.tx.as_slice();
 
         if tx_raw[0] >= 0xc0 && tx_raw[0] <= 0xfe {
-            let signed_legacy_result = TxLegacy::decode_signed_fields(tx_raw);
+            let mut signed_legacy_result = TxLegacy::decode_signed_fields(tx_raw);
+            // If we fail to decode with the strict RLP from reth,
+            // and we don't have a raw.sender which suggests a native signed trx
+            // then try the telos legacy decode without passing a signature
+            if signed_legacy_result.is_err() && raw.sender.is_none() {
```

Comment:
This should be the same without having 2 if checks for signed_legacy_result error
```suggestion
if signed_legacy_result.is_err() {
            if raw.sender.is_none() {
                signed_legacy_result =
                    TxLegacy::decode_telos_signed_fields(&mut raw.tx.clone().as_slice(), None);
            }
            let address = Address::from(
                raw.sender
                    .expect("Failed to get address from sender in unsigned transaction, signed_legacy_result is Err")
                    .data,
            );
            let sig = make_unique_vrs(block_hash, address, trx_index);
            let unsigned_legacy = TxLegacy::decode_telos_signed_fields(
                &mut raw.tx.clone().as_slice(),
                Some(sig),
            )?;
            let envelope = TxEnvelope::Legacy(unsigned_legacy);
            return Ok(TelosEVMTransaction { envelope, receipt });
        }
```

Reply:
This does change the behavior, if we decode_telos_signed_fields with a None signature successfully then we don't want to enter the second if block.